presto-main/src/main/java/io/prestosql/execution/MemoryRevokingScheduler.java

```diff
@@ -244,34 +256,105 @@ public Long mergeResults(List<Long> childrenResults)
     private void requestRevoking(MemoryPool memoryPool, Collection<SqlTask> sqlTasks, long remainingBytesToRevoke)
     {
         AtomicLong remainingBytesToRevokeAtomic = new AtomicLong(remainingBytesToRevoke);
-        sqlTasks.stream()
-                .filter(task -> task.getTaskStatus().getState() == TaskState.RUNNING)
-                .filter(task -> task.getQueryContext().getMemoryPool() == memoryPool)
-                .sorted(ORDER_BY_CREATE_TIME)
-                .forEach(task -> task.getQueryContext().accept(new VoidTraversingQueryContextVisitor<AtomicLong>()
-                {
-                    @Override
-                    public Void visitQueryContext(QueryContext queryContext, AtomicLong remainingBytesToRevoke)
+        Map<Long, List<SqlTask>> revocableMemoryMapping = new HashMap<>();
+
+        if (prioritizeLargerSpiltsMemoryRevoke) {
+            sqlTasks.stream()
```

Comment:
Mention in comment this approach disregards create time.

Reply:
addressed